massage/resynth/guitar_strummer.py

```diff
@@ -0,0 +1,275 @@
+            energy_interp_ch = scipy.interpolate.interp1d(
+                env_times, envelope[0], bounds_error=False,
+                fill_value='extrapolate')
+            energy_interp.append(energy_interp_ch)
+
+        # Generating midi for synthesis and also jams file
+        midi_data = self._generate_chord_midi(
+            chord_sequence, onsets, offsets, velocities, program,
+            energy_interp, VOICING_FILE)
+        # Convert midi_data -> jams file
+        jams_out = util.midi_to_jams(midi_data) # audio duration
+
+        # Resynth
+        y = midi_data.fluidsynth(sf2_path=sound_font, fs=fs)
+        # np.savez('mock_y', y)
+        if len(y) == 0:
+            print("y is none")
+            return
+
+        # Normalize
+        if np.max(np.abs(y)) > 0:
+            y = y / np.max(np.abs(y))
+
+        y_len = len(y)
+        env_len = len(envelope[0])
+        if y_len > env_len:
+            y_stereo = np.array(
+                [envelope[0] * y[:env_len], envelope[1] * y[:env_len]])
+        else:
+            y_stereo = np.array(
+                [envelope[0][:y_len] * y, envelope[1][:y_len] * y])
+
+        return y_stereo, jams_out
+
+    @classmethod
+    def get_id(cls):
+        """Method to get the id of the pitch tracker
+
+        Returns
+        -------
+        str : 'guitar_strummer'
+        """
+        return 'guitar_strummer'
+
+    def _get_strum(self, start_t, end_t, chord, voicings, prev_voicing,
+                   backwards=False, velocity=None):
+        """
+        Parameters
+        ----------
+        start_t: float
+            starting time of the strum
+        end_t: float
+            ending time of the strum
+        chord: str
+            'Ab:maj6', 'D#:min', etc
+        voicings: dict
+            {chord_string: list of voicing}
+            voicing is a list of int
+        prev_voicing: list of int
+            a list of midi numbers to represent the previous voicing
+        backwards: bool, default = False
+            - False : strumming down
+            - True  : strumming up
+        velocity : int
+            midi velocity value of the strum to be created
+
+        Returns
+        -------
+        strum : list of pretty_midi.Note
+            a list of pretty_midi.Note object with small delays to simulate
+            strumming
+        current_voicing : list of int
+            a list of the pitches in midi number to specify voicing
+        """
+        strum = []
+
+        notes = util.choose_voicing(chord, voicings, prev_voicing)
+        strum_duration = end_t - start_t
+
+        if strum_duration > 2.5:
+            end_t = start_t + 2.5
+            strum_duration = 2.5
+
+        current_voicing = notes
+        shifted_start = start_t
+
+        # USING FIXED VELOCITY RANGE HERE BECAUSE WE'RE ADAPTING
+        # THE OVERALL ENVELOPE LATER!
+        if velocity is None:
+            velocity = np.random.choice(range(70, 100))
```

Comment:
similar to above, 70 and 100 should be parameters `max_velocity` and `min_velocity` with defaults

Reply:
This is a inner part of the code that I want to leave hard coded.